### ISSUE TITLE
Improve CI workflows to use even more test bundling advantages

### ISSUE THREAD

**Issue by fylyppo** [leancodepl/patrol#1284]
### Overview

As we're getting soon to fully implement test bundling and #1240 is done, we want to improve our CI workflows to have:

- [ ] Test sharding (running tests in parallel) for iOS simulators (low priority):

We used to test [bluepill](https://github.com/MobileNativeFoundation/bluepill) for a while but we have two issues:
* we need to set location to all simulators (because we don't know which simulator this test will go to) before run tests 
* we need to standarize test outputs - after tests with bluepill I couldn't find XCRESULT file which I provide to job summary as a MD report. There were some JUnit XML and HTML files but they don't fit Action for detailed report we use right now. [Check here](https://github.com/leancodepl/patrol/runs/13468254072) example JUnit XML parsed by [Test Reporter](https://github.com/marketplace/actions/test-reporter). Maybe if we exclude `RunnerTests.xctest` this table will work, Idk.
---
- [ ] Test sharding for Android devices in Firebase Test Lab (mid priority):

- We can try this solution [Flank](https://flank.github.io/flank/)
- Still remember about standarizing test outputs. They provide HTML report and JUnit XML report
---
- [ ] Manage signing for Xcode release builds (we need `release` mode to run tests on FTL) (high priority):

There are few solutions, examples:

-  [Fastlane solution to automatically manage signings](https://docs.fastlane.tools/codesigning/getting-started/#using-xcodes-code-signing-feature)
- [Github manual solution](https://docs.github.com/en/actions/deployment/deploying-xcode-applications/installing-an-apple-certificate-on-macos-runners-for-xcode-development)
---
- [ ] Test sharding for iOS devices in Firebase Test Lab (mid priority):

- First of all, we need to configure code signing
- We can use also [Flank](https://flank.github.io/flank/) so test outputs could be more standarized as for Android devices
---
- [ ] Upload artifact per shard and set artifact name to one that points out what tests have been run in specific artifact (mid/high priority):
 - FTL and emulator.wtf provide test outputs with dirs called like "0,1,2,3,4,[...]" so we don't know which tests are in which directory 
 - It'll be handy if we somehow name artifacts by test names which are included in specific shard. Then we can upload artifact per shard
---
- [ ] Improve job summaries so as to have good-looking, detailed and similar for both platforms (Android, iOS) markdown summaries (high priority):
 - Firstly, we HAVE to standarize all test outputs (from each workflow) then we can think of improving test reports which are provided to job summary as MD
 - It'll be good to have the same format for table with results for every workflow. The idea is to have parser for XCRESULT and JUnit XML to one MD format. Then we can provide this report to run summary
 - Maybe we can provide links to videos from each test?
---
One obvious thing to do:

- [ ] Remove Github Actions job or step duplication

- With [this doc](https://docs.github.com/en/actions/using-workflows/reusing-workflows)
---
And we have also one tough nut to crack in our minds (but it could be done in the further future):

- [ ] Speed up Xcode builds

 - We could investigate solution from this article https://dev.sweatco.in/speeding-up-xcode-archive-builds/

**Comment by bartekpacia** [comment]
@fylyppo What's the status here?

**Comment by bartekpacia** [comment]
@fylyppo What's the status here?

**Comment by fylyppo** [comment]
outdated, closing

**Comment by bartekpacia** [comment]
:)
